src/engine/renderer/tr_shader.cpp

```diff
@@ -3678,6 +3701,46 @@ static bool ParseShader( const char *_text )
 		}
 	}
 
+	// DarkPlaces compatibility
+	// It looks for extra maps based on their suffixes
+	int c;
+	shaderStage_t colorMapStage;
+
+	for ( c = 0; c < s; c++ ) {
+		if ( stages[c].type == stageType_t::ST_COLORMAP && stages[c].textureName != nullptr ) {
+			colorMapStage = stages[c];
+			break;
+		}
+	}
+
+	if ( colorMapStage.textureName[0] != '\0' )
```

Comment:
If the immediately preceding for loop does not find anything, this reads from an uninitialized value.

Reply:
good catch!